vm/devices/net/netvsp/src/lib.rs

```diff
@@ -1971,7 +1972,11 @@ fn parse_packet<'a, T: RingMem>(
         IncomingPacket::Data(data) => data,
         IncomingPacket::Completion(completion) => {
             let data = if completion.transaction_id() == VF_ASSOCIATION_TRANSACTION_ID {
```

Comment:
This block seems pretty complicated now and has a lot of rightward drift - is there any reasonable way to make this a subfunction, that maybe wouldn't have early returns? Something like `parse_incoming_packet(...) -> Result<PacketData, Err>`?

Reply:
Tested manually on my lab machine, but let me look into a unit test.